larreco/RecoAlg/PMAlg/PmaElement3D.cxx

```diff
@@ -0,0 +1,342 @@
+/**
+ *  @file   PmaElement3D.cxx
+ *
+ *  @author D.Stefan and R.Sulej
+ * 
+ *  @brief  Implementation of the Projection Matching Algorithm
+ *
+ *          Base for 3D segments and track nodes. See PmaTrack3D.h file for details.
+ *          See PmaTrack3D.h file for details.
+ */
+
+#include "larreco/RecoAlg/PMAlg/PmaElement3D.h"
+#include "larreco/RecoAlg/PMAlg/SortedObjects.h"
+#include "larreco/RecoAlg/PMAlg/Utilities.h"
+
+#include "messagefacility/MessageLogger/MessageLogger.h"
+
+// Impact factors on the objective function:  U     V     Z
+float pma::Element3D::fOptFactors[3] =     { 0.2F, 0.8F, 1.0F };
+
+pma::Element3D::Element3D() :
+	fTPC(-1), fCryo(-1),
+	fFrozen(false),
+	fHitsRadius(0)
+{
+	fNThisHitsEnabledAll = 0;
+	for (unsigned int i = 0; i < 3; i++)
+	{
+		fNHits[i] = 0;
+		fNThisHits[i] = 0;
+		fSumHitsQ[i] = 0.0;
+	}
+}
+
+size_t pma::Element3D::NEnabledHits(unsigned int view) const
+{
+	size_t n = 0;
+	for (size_t i = 0; i < fAssignedHits.size(); i++)
+		if (fAssignedHits[i]->IsEnabled() &&
```

Comment:
Given the index value `i` is not needed for this loop, I'd suggest switching to the `for( auto const& hit: fAssignedHits)` form.
In addition, I suggest adding `{...}` to clarify scope. I know this code presently works, but if it is ever modified it is very easy to think the modification is within the `for` or `if` scope but be wrong.